### PR TITLE
DefaultHttp2HeadersTest updates

### PR DESCRIPTION
**Pull request by Scottmitch**
Motivation:
Recently a bug was found in DefaultHttp2Headers where the state of the headers could be corrupted due to the extra tracking to make pseudo headers first during iteration. Unit tests did not catch this bug.

Modifications:
- Update unit tests to cover more methods

Result:
Unit tests for DefaultHttp2Headers have better code coverage.

### CODE REVIEW

**Comment by Scottmitch** [comment]
@nmittler @louiscryan @normanmaurer - FYI

File: codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2HeadersTest.java [hunk 1]
```diff
@@ -68,6 +67,58 @@ public void pseudoHeadersWithRemovePreservesPseudoIterationOrder() {
         }
     }
 
+    @Test
+    public void pseudoHeadersWithClearDoesNotLeak() {
```
Comment (normanmaurer) [inline codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2HeadersTest.java]: Other tests are prefixed with test. Maybe do here as well?

Comment (Scottmitch) [inline codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2HeadersTest.java]: +1

**Comment by Scottmitch** [comment]
@normanmaurer - pushed new code. ptal.

**Comment by normanmaurer** [comment]
LGTM

**Comment by Scottmitch** [comment]
Cherry-picked 4.1 (2f81364) master (b21fe4e)

**Pull request closed without merging**